packages/jest-haste-map/src/index.ts

```diff
@@ -739,17 +737,18 @@ export default class HasteMap extends EventEmitter {
   /**
    * Creates workers or parses files and extracts metadata in-process.
    */
-  private _getWorker(options?: {forceInBand: boolean}): WorkerInterface {
+  private _getWorker(options?: {
+    forceInBand: boolean;
+  }): JestWorkerFarm<HasteWorker> | HasteWorker {
     if (!this._worker) {
       if ((options && options.forceInBand) || this._options.maxWorkers <= 1) {
         this._worker = {getSha1, worker};
       } else {
-        // @ts-expect-error: assignment of a worker with custom properties.
-        this._worker = new Worker(require.resolve('./worker'), {
+        this._worker = new WorkerFarm(require.resolve('./worker'), {
```

Comment:
not `createWorkerFarm`?

Reply:
Yep. Still struggling with this one.

Reply:
@SimenB I was trying again to make this work, but cannot figure out how to make async function work in sync context. Probably some `.then` should be used to unwrap the promise. Might work, but this function already returns `worker()` that returns a Promise:

https://github.com/facebook/jest/blob/2f4340cb1da435a9b147e238f7034f99653b8070/packages/jest-haste-map/src/index.ts#L649-L658

I was trying to play with `.then`s and to refactor this code to use `async` / `await`. Unfortunately it is too complex for me. Can’t make it work. Someone more experienced should take a look.

Reply:
> cannot figure out how to make async function work in sync context.

Why/where does it have to work in sync context? If it's just https://github.com/facebook/jest/blob/8ef4d86f14954277394c7e4fb708697dfa92e80c/packages/jest-haste-map/src/index.ts#L743, making that return a `Promise` seems fine?

Reply:
Right, `_getWorker` can return a Promise. It is called and `then` is used to handle a Promise return by the worker:

https://github.com/facebook/jest/blob/8ef4d86f14954277394c7e4fb708697dfa92e80c/packages/jest-haste-map/src/index.ts#L652-L661

The `return` above is is the return of `_processFile` method, that has a return type of `Promise<void> | null`:

https://github.com/facebook/jest/blob/8ef4d86f14954277394c7e4fb708697dfa92e80c/packages/jest-haste-map/src/index.ts#L451-L457

 This `null` part is where sync fun begins. It just must be `Promise<void> | null`, not something like `Promise<void | null>`, because of these lines:

https://github.com/facebook/jest/blob/8ef4d86f14954277394c7e4fb708697dfa92e80c/packages/jest-haste-map/src/index.ts#L703-L709

I was playing with the filter above for some time with no luck. Might be some deeper refactor is needed.